app/Http/Controllers/Admin/RoomLayoutController.php

```
@@ -73,26 +75,43 @@ public function edit(Room $room)
-
-                    if (! $isTempId && ! $isExistingRoomBlock) {
-                        $fail('The selected block id is invalid for this room.');
+            'markerBlocks' => 'sometimes|array',
+            'markerBlocks.*.id' => ['nullable', $ownedId($room->markerBlocks(), 'The selected marker block id is invalid for this room.')],
+            'markerBlocks.*.type' => 'required|string|in:entrance,comment',
+            'markerBlocks.*.name' => 'required|string|max:255',
+            'markerBlocks.*.position_x' => ['required', 'integer', 'min:-1', 'max:11'],
+            'markerBlocks.*.position_y' => ['required', 'integer', 'min:-1', 'max:7'],
+            'markerBlocks.*.rotation' => 'nullable|integer|in:0,90,180,270',
+            'markerBlocks.*' => [
+                function (string $attribute, $value, $fail) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$attribute'. (undefined)

(UnusedFormalParameter)